### PR TITLE
CORE-853: check tx signature/witness for wallet inclusion

### PR DESCRIPTION
**Pull request by EBGToo**
Also modifies BRWalletManager.c to address possible P2P issues seen in IOS related to 'isDeleted'

### CODE REVIEW

**Comment by EBGToo** [comment]
@voisine Can you review `BRWalletIsResolved` (and associated `_BRWalletContainsTxInput`)?  A transaction is resolved if all inputs on wallet have known dependent transactions.

@michael-brd Can you review again? I added an 'isResolved' flag to BRTransactionWithState.  Until resolved events are not generated.  In 'txAdded' all known BRWalletManager transactions are reviewed; any that are newly resolved have their {ADDED, UPDATED} events generated.

**Comment by EBGToo** [comment]
I've tested to show that API and P2P modes both sync successfully with the 'transaction that arrives early' not being announced until its dependent transaction arrives.  Aaron's earlier change that recognized the 'transaction that arrives early' adds the transaction to wallet; the 'isResolved' change avoids that added transaction from having a zero amount.

File: bitcoin/BRWalletManager.c [hunk 1]
```diff
@@ -1364,6 +1364,7 @@ BRWalletManagerCreateTransaction (BRWalletManager manager,
     pthread_mutex_unlock (&manager->lock);
 
     if (NULL != txnWithState) {
+        BRTransactionWithStateSetResolved (txnWithState);  // Always resolved if created
```
Comment (michael-brd) [inline bitcoin/BRWalletManager.c]: Modifications should be in the lock, I believe. To be honest, the gets should be there too (a mistake on my part).

Comment (EBGToo) [reply bitcoin/BRWalletManager.c]: Addressed w/ the 'gets' too.

**Comment by EBGToo** [comment]
Tested sending a transaction.  As expected 'isResolved' and events are generated.

**Pull request merged**